app/src/main/java/com/pvp/app/service/DrinkReminderWorker.kt

```diff
@@ -0,0 +1,123 @@
+package com.pvp.app.service
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.pvp.app.api.Configuration
+import com.pvp.app.api.NotificationService
+import com.pvp.app.api.SettingService
+import com.pvp.app.api.UserService
+import com.pvp.app.model.Notification
+import com.pvp.app.model.NotificationChannel
+import com.pvp.app.model.Setting
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import java.time.Duration
+import java.time.LocalTime
+
+@HiltWorker
+class DrinkReminderWorker @AssistedInject constructor(
+    @Assisted context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val notificationService: NotificationService,
+    private val settingService: SettingService,
+    private val userService: UserService,
+    private val configuration: Configuration
+) : CoroutineWorker(context, workerParams) {
+
+    companion object {
+        const val WORKER_NAME = "com.pvp.app.service.DrinkReminderWorker"
+    }
+
+    private val coroutineScope = CoroutineScope(Dispatchers.Main)
+    private val scheduledNotifications = mutableListOf<Notification>()
+
+    override suspend fun doWork(): Result {
+        initializeReminderListener()
+
+        return Result.success()
+    }
+
+    private suspend fun initializeReminderListener() {
+        val stateFlow = settingService
+            .get(Setting.Notifications.CupVolumeMl)
+            .combine(userService.user) { volume, user ->
+                DrinkReminderState(
+                    cupVolume = volume,
+                    userMass = user?.mass ?: 0
+                )
+            }
+            .stateIn(
+                scope = coroutineScope,
+                started = SharingStarted.WhileSubscribed(5000),
+                initialValue = DrinkReminderState()
+            )
+
+        stateFlow.collect { state ->
+            if (state.userMass != 0) {
+                cancelScheduledNotifications()
+
+                scheduleNotifications(
+                    mass = state.userMass,
+                    cupVolume = state.cupVolume
+                )
+            }
+        }
+    }
+
+    private fun scheduleNotifications(
+        mass: Int,
+        cupVolume: Int
+    ) {
+        val recommendedIntake = mass * 30
+        val nbOfReminders = recommendedIntake / cupVolume
```

Comment:
Don't use shortened abbreviations. Just write the full meaning (word) or use simpler naming. This could be just `reminders` or `count` and woul be quite understandable.